internal/local/structure.go

```diff
@@ -0,0 +1,222 @@
+		if err != nil {
+			return err
+		}
+		out[file.Name()] = o
+		return nil
+	}); err != nil {
+		return nil, err
+	}
+
+	if len(out) == 0 {
+		return nil, nil
+	}
+	return out, nil
+}
+
+func parseFunctions(rootDir string) ([]map[string]interface{}, error) {
+	if ok, err := fileExists(rootDir); err != nil {
+		return nil, err
+	} else if !ok {
+		return nil, nil
+	}
+
+	var out []map[string]interface{}
+
+	dw := directoryWalker{path: rootDir, onlyDirs: true}
+	if walkErr := dw.walk(func(file os.FileInfo, path string) error {
+		if strings.Contains(path, nameNodeModules) {
+			return nil // skip node_modules
+		}
+
+		config, configErr := parseJSONConfig(path)
+		if configErr != nil {
+			return configErr
+		}
+
+		src, srcErr := parseJavascriptSource(path)
+		if srcErr != nil {
+			return srcErr
+		}
+
+		out = append(out, map[string]interface{}{
+			NameConfig: config,
+			NameSource: src,
+		})
+		return nil
+	}); walkErr != nil {
+		return nil, walkErr
+	}
+
+	return out, nil
+}
+
+func parseGraphQL(rootDir string) (GraphQLStructure, bool, error) {
+	dir := filepath.Join(rootDir, NameGraphQL)
+
+	if ok, err := fileExists(dir); err != nil {
+		return GraphQLStructure{}, false, err
+	} else if !ok {
+		return GraphQLStructure{}, false, nil
+	}
+
+	config, configErr := parseJSONConfig(dir)
+	if configErr != nil {
+		return GraphQLStructure{}, false, configErr
+	}
+
+	customResolvers, customResolversErr := parseJSONFiles(filepath.Join(dir, NameCustomResolvers))
+	if customResolversErr != nil {
+		return GraphQLStructure{}, false, customResolversErr
+	}
+
+	return GraphQLStructure{config, customResolvers}, true, nil
+}
+
+func parseJavascript(rootDir string, file File) (string, error) {
+	src, err := ioutil.ReadFile(filepath.Join(rootDir, file.String()))
+	if err != nil {
+		return "", err
+	}
+	return string(src), nil
+}
+
+func parseJavascriptSource(rootDir string) (string, error) {
+	return parseJavascript(rootDir, FileSource)
+}
+
+func parseJSON(path string) (map[string]interface{}, error) {
+	data, dataErr := readFile(path)
+	if dataErr != nil {
+		return nil, dataErr
+	}
+
+	var out map[string]interface{}
+	if err := unmarshalJSON(data, &out); err != nil {
+		return nil, err
+	}
+	return out, nil
+}
+
+func parseJSONConfig(rootDir string) (map[string]interface{}, error) {
+	return parseJSON(filepath.Join(rootDir, FileConfig.String()))
+}
+
+func parseJSONFiles(rootDir string) ([]map[string]interface{}, error) {
+	if ok, err := fileExists(rootDir); err != nil {
+		return nil, err
+	} else if !ok {
+		return nil, nil
+	}
+
+	out := make([]map[string]interface{}, 0)
+
+	dw := directoryWalker{path: rootDir, onlyFiles: true}
+	if walkErr := dw.walk(func(file os.FileInfo, path string) error {
+		o, err := parseJSON(path)
+		if err != nil {
+			return err
+		}
+
+		out = append(out, o)
+		return nil
+	}); walkErr != nil {
+		return nil, walkErr
+	}
+
+	return out, nil
+}
+
+func parseJSONSecrets(rootDir string) (*SecretsStructure, error) {
```

Comment:
[nit] and for all the other funcs - can we just call this parseSecrets? I can't imagine we ever support anything other than JSON. Plus, with the abstraction of this func the caller doesn't need to be aware of what format the data is in.

Reply:
Also - I would probably opt for this to return an io.Reader or []byte and then pull the json.Marshal up into the calling func. this could simply become`readSecrets`

Reply:
yeah, i can change that name to remove `JSON` from it, but im going to stick with read/marshaling alongside each other in these funcs.  its logic that's always executed together within this CLI context and eventually.  i also like this breakdown for the inevitable `functions` unit tests I'm going to be writing, making sure we're parsing and collecting data from that directory properly